js/private/js_binary.bzl

```diff
@@ -168,6 +169,19 @@ _ATTRS = {
         """,
         default = True,
     ),
+    "copy_data_to_bin": attr.bool(
+        doc = """When True, data files and the entry_point file are copied to the Bazel output tree before being passed
+        as inputs to runfiles.
+
+        This default to False. It should only be needed if the program manages to skirt the node js patches and leave
+        its runfiles and/or sandbox. In that case, setting this to True will prevent the program from following symlinks
+        into the source tree and it will end up in the output tree instead where node_modules and other inputs the
+        program may need will be.
+
+        Mocha, for example, has been observed to escape its runfiles & sandbox: https://github.com/aspect-build/rules_js/pull/353.
```

Comment:
Do we actually understand why? It would be nice to have a full explanation.

I'm also curious if things like mocha are doing something "wrong" and wonder if it would be worth reporting it to them. I think we (or at least I) need to understand what's going on here more before thinking about that.

Reply:
We don't yet. One of us needs to take the time to see what mocha is doing under the hood.